source/ConversationPanel.cpp

```diff
@@ -343,21 +343,22 @@ void ConversationPanel::Goto(int index, int choice)
 void ConversationPanel::Exit()
 {
 	GetUI()->Pop(this);
-	// If this is a conversation offered from boarding or assisting an NPC,
-	// being forced to leave via LAUNCH, FLEE, or DEPART destroys it. If it
-	// is a hostile NPC and not destroyed, and its mission (e.g. "leave me
-	// alone, here's X as payment") was not accepted, open a BoardingPanel
-	// to allow plundering it - unless you're dead.
-	if(node != Conversation::DIE && player.BoardingShip())
+	// If this is a conversation offered from an NPC, ending it via LAUNCH,
+	// FLEE, or DEPART destroys the NPC. For hostile NPCs that are boarded
+	// and not being destroyed, show the BoardingPanel unless the player is
+	// being killed or the conversation is exiting via ACCEPT.
+	if(node != Conversation::DIE && node != Conversation::EXPLODE && ship)
```

Comment:
Note: the `ship` pointer could be a ship you just scanned that triggered a conversation, in which case making it explode seems weird.

Reply:
`"We've sabotaged this <npc> with explosives, and they're set to go off when you scan their cargo"`

Reply:
> "We've sabotaged this \<npc\> with explosives, and they're set to go off when you scan their cargo"

[gets some ideas]